services/housefy/src/main/java/com/dech/housefy/service/impl/SalesmanImpl.java

```
@@ -42,11 +52,79 @@ public List<SalesmanDTO> uploadCsvSalesmen(MultipartFile file) {
         return salesmenDTO;
     }
 
+    @Override
+    public SalesmanDTO createSalesman(SalesmanDTO salesmanDTO) {
+        var exist = salesmanRepository.existsByNameAndLastName(salesmanDTO.getName(),salesmanDTO.getLastName());
+        if (exist == false) {
+            var salesman = new Salesman();
+            salesman.setName(salesmanDTO.getName());
+            salesman.setLastName(salesmanDTO.getLastName());
+            salesman.setPhoneNumber(salesmanDTO.getPhoneNumber());
+
+            logger.info("salesman to create {}", salesman);
+            var salesmanCreated = this.salesmanRepository.save(salesman);
+            var newSalesmanDTO = this.entityToDto(salesmanCreated);
+
+            return newSalesmanDTO;
+        }
+        throw new DuplicateDataException("There is a salesman created with the name: " + salesmanDTO.getName() + " " + salesmanDTO.getLastName());
+    }
+
+    @Override
+    public SalesmanDTO getSalesmanById(String id) {
+        var salesman = this.salesmanRepository.findById(id).orElseThrow();
```

you should throw an exception like DataNotFounfException